### DISCUSSION TITLE
Regarding EB boundary conditions

### DISCUSSION THREAD

**Discussion by RSuryaNarayan** [AMReX-Combustion/PeleC#313] [Q&A] [answered]
Hello everyone, I have a question on the boundary conditions available currently for treating EB surfaces. 
I see that there is an input: 
```
pelec.eb_isothermal =1
pelec.eb_boundary_T = 1000 # this is the EB surface temperature
```
What is the default thermal treatment for the EB surface (i.e. if none of the above are provided?). 
Is it adiabatic?

**Comment by hsitaram** [comment, marked as answer]
Yes, it's an adiabatic wall by default.

**Reply by RSuryaNarayan** [reply]
@hsitaram thanks !

**Reply by RSuryaNarayan** [reply]
I have a similar question on NSCBC. The documentation says NSCBC is turned on [automatically for subsonic inflows into the domain](https://amrex-combustion.github.io/PeleC/BoundaryConditions.html#:~:text=By%20default%2C%20the%20GC-NSCBC%20method%20is%20activated%20for%20all%20subsonic%20flow%20boundaries.%20It). 
Does this still hold? 
If yes, does this get overwritten when I impose a "Hard" BC for the subsonic inlet and an "FOExtrap" for the outlet? 
Hence, do we additionally have to code in the response of the outlet/inlet when defining subsonic inflow?

**Reply by adityaIyerramesh98** [reply]
Hey Surya,
Correct me If am off-track. I believe, it ain't overwrites when inflow and outflow BCs (Hard for ghost cells and First order Extrapol) is carried out for 1D flame. Referring to the same docs you mentioned above, I got this line which states,

`! When optional arguments are present, GC-NSCBC is activated
  ! Generic values are auto-filled for numerical parameters,
  ! but should be set by the user for each BC
  ! Note that in the impose_NSCBC_xD.f90 routine, not all parameters are used in same time
  if (present(bc_type).and.present(bc_params).and.present(bc_target)) then

    flag_nscbc = 1
    relax_U = 0.5d0 ! For inflow only, relax parameter for x_velocity
    relax_V = 0.5d0 ! For inflow only, relax parameter for y_velocity
    relax_W = 0.5d0 ! For inflow only, relax parameter for z_velocity
    relax_T = 0.2d0 ! For inflow only, relax parameter for temperature
    beta = 0.2d0  ! Control the contribution of transverse terms
    sigma_out = 0.25d0 ! For outflow only, relax parameter
    which_bc_type = Interior ! This is to ensure that nothing will be done if the user don't set anything
  endif`

On paging through comments for the above snippet, It depicts "user has to set the code before executing BCs" which further draws a vindication that we have to make additions/inclusions beforehand to the response code of GC-NSCBC routine.
Thanks!

**Reply by RSuryaNarayan** [reply]
This references the FORTRAN routines that are currently deprecated

**Reply by adityaIyerramesh98** [reply]
Understood. May I ask, are you computing in 2D? 
The above snippet is purely illustrative and I believe the documentation is still in the pipeline. Saliency vis-à-vis overwriting in response code for the FORTRAN routines of GC-NSCBC is concerned, the user has to manually fill in the code beforehand, let alone till the automation sets in. Got anything to add?

CC: @drummerdoc , @hsitaram and @emotheau

**Reply by marchdf** [reply]
Hi all, I created PR #319 to the docs for clarifying that GC-NSCBC has not yet been ported to C++ from Fortran. Thanks for bringing it to our attention.

**Reply by RSuryaNarayan** [reply]
@marchdf thanks, so currently we have only first order extrapolation as a way to accomplish a perfectly non-reflecting BC?

**Reply by marchdf** [reply]
So you can try first order extrapolation and hope to get lucky. If that doesn't work, you can try to specify the ghost cell values in a way that reduces reflections. A good example of that is what I do in EB-C10: https://github.com/AMReX-Combustion/PeleC/blob/development/Exec/RegTests/EB-C10/prob.H#L91 which attempts to follow Blazek's subsonic outflow condition. I've had good luck with that but it's by no means full-proof or perfect.

**Reply by RSuryaNarayan** [reply]
Thanks! FOExtrap worked for the acoustic pulse in channel case (EB-C9). I also have cases where I need a part of the wall as a non-reflecting wall and partly as an inflow. I think Blazek's method should work as a replacement for FOExtrap here.

**Reply by marchdf** [reply]
I guess it's not too surprising that it would work for that case. It's a weak acoustic pulse. Nothing crazy going on. We've seen FOExtrap fail for things that have fluctuating velocities (e.g. turbulence...)
